lib/ecto_adapters_dynamodb.ex

```diff
@@ -735,14 +674,60 @@ defmodule Ecto.Adapters.DynamoDB do
     end
   end
 
+
+
+  # Support for tables with a hash+range key.
+  #
+  #  * If the schema has both keys declared (using the `primary_key: true`) the filters are already correct
+  #  * If :range_key is specified with a value it is added to filters
+  #  * If :range_key is not specified, and the table does have a range key, attempt to find it with a DynamoDB query
+  #
+  def maybe_update_filters_for_range_key(table, schema_meta, filters, opts) do
+      with primary_key_length <- length(schema_meta.schema.__schema__(:primary_key)) do
+        case opts[:range_key] do
+          # Use primary keys declared in schema
+          nil when primary_key_length == 2 ->
+            filters
+          nil ->
+            {:primary, key_list} = Ecto.Adapters.DynamoDB.Info.primary_key!(table)
+            if (length key_list) > 1 do
+              updated_opts = opts ++ [projection_expression: Enum.join(key_list, ", ")]
+              filters_as_strings = for {field, val} <- filters, do: {Atom.to_string(field), {val, :==}}
+              fetch_result = Ecto.Adapters.DynamoDB.Query.get_item(table, filters_as_strings, updated_opts)
+              items = case fetch_result do
+                %{"Items" => fetch_items} -> fetch_items
+                %{"Item" => item}         -> [item]
+                _                         -> []
+              end
+
+              if items == [], do: raise "__MODULE__.update error: no results found for record: #{inspect filters}"
+              if (length items) > 1, do: raise "__MODULE__.update error: more than one result found for record: #{inspect filters} Please consider using the adapter's :range_key custom inline option (see README)."
```

Comment:
Little nitpick: (`__MODULE__` should be `#{inspect __MODULE__}` notwithstanding :) if this is to become one function, the error text should state the correct caller, `.update` vs `.delete`.

Reply:
Thanks, @alhambra1 

Do you prefer

1.  to keep this code inline within the `update` and `delete` functions? Or,
2. keep the extracted function? 

If (2) we could pass the name of the operation as an argument to be included in the exception message.